src/pages/index.js

```diff
@@ -11,6 +11,32 @@ const App = () => {
   const [data, setData] = useState(null);
   const [error, setError] = useState(null);
 
+  useEffect(() => {
+    if (props.username) {
+      setUsername(cleanUsername(props.username));
+      setLoading(true);
+      setError(null);
+      setData(null);
+
+      fetchData(cleanUsername(props.username))
```

Comment:
This part is a duplicate of existing functionality. Is it possible for you to use the same code for both? (like extracting the fetch into its own function)